samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/Application.java

```diff
@@ -1,2 +1,73 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+
+package com.microsoft.bot.sample.core;
+
+import com.microsoft.bot.builder.Bot;
+import com.microsoft.bot.integration.AdapterWithErrorHandler;
+import com.microsoft.bot.integration.BotFrameworkHttpAdapter;
+import com.microsoft.bot.integration.Configuration;
+import com.microsoft.bot.integration.spring.BotController;
+import com.microsoft.bot.integration.spring.BotDependencyConfiguration;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+
+//
+// This is the starting point of the Sprint Boot Bot application.
+//
+@SpringBootApplication
+
+// Use the default BotController to receive incoming Channel messages. A custom
+// controller could be used by eliminating this import and creating a new
+// org.springframework.web.bind.annotation.RestController.
+// The default controller is created by the Spring Boot container using
+// dependency injection. The default route is /api/messages.
+@Import({BotController.class})
+
+/**
+ * This class extends the BotDependencyConfiguration which provides the default
+ * implementations for a Bot application.  The Application class should
+ * override methods in order to provide custom implementations.
+ */
+public class Application extends BotDependencyConfiguration {
+    public static void main(String[] args) {
+        SpringApplication.run(Application.class, args);
+    }
+
+    /**
+     * Returns the Bot for this application.
+     *
+     * <p>
+     *     The @Component annotation could be used on the Bot class instead of this method
+     *     with the @Bean annotation.
+     * </p>
+     *
+     * @return The Bot implementation for this application.
+     */
+    @Bean
+    public Bot getBot(
+        Configuration configuration,
+        ConversationState conversationState,
+        UserState userState,
+        FlightBookingRecognizer flightBookingRecognizer,
+        BookingDialog bookingDialog,
+        MainDialog dialog
+    ) {
+        return new CoreBot(conversationState, userState, flightBookingRecognizer, bookingDialog, new MainDialog(configuration));
```

Comment:
Why are we generating a new `MainDialog` when you have it as parameter? See [46 sample](https://github.com/microsoft/botbuilder-java/blob/main/samples/46.teams-auth/src/main/java/com/microsoft/bot/sample/teamsauth/Application.java#L56)

```suggestion
        return new CoreBot(conversationState, userState, flightBookingRecognizer, bookingDialog, dialog);
```

Reply:
Also this should return a `DialogAndWelcomeBot`. See [C#](https://github.com/microsoft/BotBuilder-Samples/blob/main/samples/csharp_dotnetcore/13.core-bot/Startup.cs#L44)